File: test_pkg/src/test_node2.cpp
#include "ros/ros.h"

#include "std_msgs/String.h"



void callback_string(const std_msgs::String::ConstPtr& msg){
	
	ROS_WARN("I heard: [%s]", msg->data.c_str());

}




int main(int argc, char **argv){

	ros::init(argc, argv, "test_node_2");

	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("/topic1", 1000, callback_string);

	ros::spin();


}

File: test_pkg/script/test_node2.py
#!/usr/bin/env python3

import rospy

from std_msgs.msg import String

def callback_string(msg):
    rospy.logwarn("I heard %s", msg.data)


if __name__ == '__main__':
    rospy.init_node("test_node_2")
    sub = rospy.Subscriber("/topic1", String, callback_string, queue_size=1000)

    rospy.spin()